catalog.py

```diff
@@ -0,0 +1,238 @@
+import math
+from enum import Enum
+from model_objects import ProductUnit
+from model_objects import ProductQuantity, SpecialOfferType, Discount
+from model_objects import Offer
+from receipt import Receipt
+
+
+class SupermarketCatalog:
+
+    def add_product(self, product, price):
+        raise Exception("cannot be called from a unit test - it accesses the database")
+
+    def unit_price(self, product):
+        raise Exception("cannot be called from a unit test - it accesses the database")
+
+
+
+    class Product:
+        def __init__(self, name, unit):
+            self.name = name
+            self.unit = unit
+
+    class ProductQuantity:
+        def __init__(self, product, quantity):
+            self.product = product
+            self.quantity = quantity
+
+    class ProductUnit(Enum):
```

Comment:
```suggestion
    class ProductUnit(Enum):
        EACH = 1
        KILO = 2

    class SpecialOfferType(Enum):
        THREE_FOR_TWO = 1
        TEN_PERCENT_DISCOUNT = 2
        TWO_FOR_AMOUNT = 3
        FIVE_FOR_AMOUNT = 4
```
Думаю наименование переменных заглавными буквами сильно выбивается на фоне всей программы